frontend/src/components/ScheduledEventsRunHistoryTableCard/ScheduledEventsRunHistoryTableCard.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { ReactElement, ReactNode, useEffect, useState } from 'react';
+import { Card } from 'react-bootstrap';
+import { useLocation } from 'react-router-dom';
+import { getScheduledEventsByPage, LIMIT } from '.././ScheduledEventsTableCard';
+import { PaginationRow } from '../PaginationRow';
+import { ScheduledEventsRunHistoryTable } from '../ScheduledEventsRunHistoryTable';
+import { fetchRunHistory } from './utils';
+
+export const ScheduledEventsRunHistoryTableCard = (): ReactElement => {
+  const location = useLocation();
+  const [historyData, setHistoryData] = useState([]);
+  const [count, setCount] = useState(0);
+  const [pageCount, setPageCount] = useState(0);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [selectedPage, setSelectedPage] = useState(0);
+
+  useEffect(() => {
+    if (location.state) {
+      fetchRunHistory(location.state.rowId).then((result) => {
+        setHistoryData(result.data);
+        setCount(result.data.length);
+        setPageCount(Math.ceil(result.data.length / 5));
+      });
+    }
+  }, [location]);
+
+  const handlePageChange = (page: { selected: number }): void => {
+    setSelectedPage(page.selected);
+    if (page.selected === selectedPage + 1) {
+      setCurrentPage((currentPage) => currentPage + 1);
+    } else if (page.selected === selectedPage - 1) {
+      setCurrentPage((currentPage) => currentPage - 1);
+    } else {
+      setCurrentPage(() => page.selected + 1);
+    }
+  };
+  const renderPagination = (): ReactNode => {
+    return count === 0 ? (
+      'No Data'
+    ) : (
+      <PaginationRow
+        limit={LIMIT}
+        count={count}
+        pageCount={pageCount}
+        onPageChange={handlePageChange}
+      />
+    );
+  };
+
+  return (
+    <div className="col-md-12">
+      {historyData && historyData.length > 0 ? (
```

Comment:
Just having historyData.length is enough since any number that isn't `0` is truthy
So it becomes `historyData && historyData.length ? (...`